### DISCUSSION TITLE
Reuse Responses - Read From a Non-existing File

### DISCUSSION THREAD

**Discussion by bytrangle** [vercel/examples#265] [Q&A]
First off, thank you very much for putting up a great example for sharing responses from `getStaticPaths` to `getStaticProps`.

I have a question regarding the use of `cache.get()` method.

`getStaticPaths` only writes to cache at build time. However, in `getStaticProps`, you try to read from cache unconditionally. If the cache file doesn't exist, then `fs.promises` will throw an error "ENOENT No such file or directory at ...".

```
const getStaticPaths = async () => {
  const products = await api.list()

  if (process.env.NEXT_PHASE === PHASE_PRODUCTION_BUILD) {
    await api.cache.set(products)
  }
  // ...
}
const getStaticProps = async ({ params }) => {
  let product = await api.cache.get(params?.id as string)

  if (!product) {
    product = await api.fetch(params?.id as string)
  }
  // ...
}
```
For now, anyone trying out this demo won't get that error because you've added a placeholder file `products.db`.

Should we **add a `catch` block to the `cache.get()` method**? Or only **read from cache if it is build time**?

**Comment by goncy** [comment]
Hey @bytrangle, thanks for the comment! As this are just meant to be used as guidance we can assume that the file will always be there. Although I feel that adding a catch would make more sense, I prefer to leave it like this to give more focus on the example related code, but of course, if we see this happens to more people we can add it!
